llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -2461,6 +2461,34 @@ InstructionCost AArch64TTIImpl::getCastInstrCost(unsigned Opcode, Type *Dst,
             FP16Tbl, ISD, DstTy.getSimpleVT(), SrcTy.getSimpleVT()))
       return AdjustCost(Entry->Cost);
 
+  if ((ISD == ISD::ZERO_EXTEND || ISD == ISD::SIGN_EXTEND) &&
+      CCH == TTI::CastContextHint::Masked && ST->hasSVEorSME() &&
+      !TLI->isTypeLegal(SrcTy) && !TLI->isTypeLegal(DstTy)) {
```

Comment:
Is this change might affect non-load extend instructions?

Reply:
That's a good question @dtemirbulatov! For non-load extends the context hint (CCH) will be TTI::CastContextHint::None, whereas I am checking for TTI::CastContextHint::Masked. That means the new code I've added only applies when we are extending a masked load.